ibis/backends/sql/dialects.py

```diff
@@ -135,6 +134,69 @@ class Generator(Hive.Generator):
             sge.Interval: _interval_with_precision,
         }
 
+        def struct_sql(self, expression: sge.Struct) -> str:
+            from sqlglot.optimizer.annotate_types import annotate_types
+
+            expression = annotate_types(expression)
+
+            values = []
+            schema = []
+            unknown_type = False
+
+            for e in expression.expressions:
+                if isinstance(e, sge.PropertyEQ):
+                    e = sge.alias_(e.expression, e.this)
+                # named structs
+                if isinstance(e, sge.Alias):
+                    if e.type and e.type.is_type(sge.DataType.Type.UNKNOWN):
+                        unknown_type = True
+                    else:
+                        schema.append(f"{self.sql(e, 'alias')} {self.sql(e.type)}")
+                    values.append(self.sql(e, "this"))
+                else:
+                    values.append(self.sql(e))
+
+            size = len(expression.expressions)
+
+            if not size or len(schema) != size:
+                if unknown_type:
```

Comment:
Is there a scenario where we have an `unknown_type` but `len(schema) == size`?

If not, I think the error can get moved up to where you are setting `unknown_type`